Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp

```diff
@@ -308,6 +308,7 @@ namespace PhysX
         AZ::TransformNotificationBus::Handler::BusConnect(entityId);
         Physics::ColliderComponentEventBus::Handler::BusConnect(entityId);
         AzFramework::BoundsRequestBus::Handler::BusConnect(entityId);
+        AzToolsFramework::EditorComponentSelectionRequestsBus::Handler::BusConnect(entityId);
```

Comment:
Both `EditorColliderComponent` and `EditorRigidBodyComponent` connect to the `zToolsFramework::EditorComponentSelectionRequestsBus` using the same entity id. An entity can have both components, so both components are handling the bus for the same entity. Is that intentional? How that does work when a caller calls a EditorComponentSelectionRequestsBus function for `entityId`? How is it controller which one responds? Does it need an entity-component pair so the caller can make sure to call the one it ones?

Reply:
It is intentional, the bus doesn't receive any calls if it is only connected to one of the components strangely which I can investigate in a follow up PR

Reply:
This is actually okay but requires the caller to use `AZ::EBusReduceResult` (see `CalculateEntityLocalBoundsUnion` as an example - this is what we do with components with world bounds as multiple components might provide an Aabb of different sizes and we need to return the aggregate/union of the bounds. So there's nothing technically incorrect with this approach.

Reply:
Usage looks like this:

```c++
AZ::EBusReduceResult<AZ::Aabb, AabbUnionAggregator> aabbResult(AZ::Aabb::CreateNull());
BoundsRequestBus::EventResult(aabbResult, entity->GetId(), &BoundsRequestBus::Events::GetLocalBounds);
```

Reply:
That's cool, I wasn't aware of that. Thanks for sharing @hultonha !

Reply:
This is actually already implemented in `PickEntity` in `AzToolsFramework/ViewportSelection/EditorSelectionUtil.cpp`

Reply:
Cool no problem @moraaar and thanks for the info @amzn-ahmadkrm 👍